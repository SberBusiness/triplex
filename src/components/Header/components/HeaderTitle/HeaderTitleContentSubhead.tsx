import React from 'react';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';

/** Свойства компонента HeaderTitleContentSubhead. */
export interface IHeaderTitleContentSubheadProps extends React.HTMLAttributes<HTMLDivElement> {}

/** Подзаголовок. */
export const HeaderTitleContentSubhead: React.FC<IHeaderTitleContentSubheadProps> = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames(className, 'cssClass[headerTitleContentSubhead]')} {...htmlDivAttributes}>
        {children}
    </div>
);

HeaderTitleContentSubhead.displayName = 'HeaderTitleContentSubhead';
