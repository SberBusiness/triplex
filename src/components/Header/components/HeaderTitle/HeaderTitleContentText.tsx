import React from 'react';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';
import {FocusTrapUtils} from '@sberbusiness/triplex/utils/focus/FocusTrapUtils';

/** Свойства компонента HeaderTitleContentText. */
export interface IHeaderTitleContentTextProps extends React.HTMLAttributes<HTMLHeadingElement> {}

/** Основной заголовок. */
export const HeaderTitleContentText = React.forwardRef<HTMLHeadingElement, IHeaderTitleContentTextProps>(
    ({children, className, ...rest}, ref) => (
        <h1
            className={classnames('cssClass[globalHeaderTitleContentText]', className)}
            //* При открытии LightBox/ModalWindow - заголовок будет зачитываться VoiceOver-ом.
            {...{[FocusTrapUtils.firstInteractionElementDataAttr]: true}}
            tabIndex={-1}
            {...rest}
            ref={ref}
        >
            {children}
        </h1>
    )
);

HeaderTitleContentText.displayName = 'HeaderTitleContentText';
