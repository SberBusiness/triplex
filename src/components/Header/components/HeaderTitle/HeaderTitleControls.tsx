import React from 'react';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';

/**
 * Свойства компонента.
 */
export interface IHeaderTitleControlsProps extends React.HTMLAttributes<HTMLDivElement> {}

/**
 * Блок с кнопками действий HeaderTitle.
 */
export const HeaderTitleControls: React.FC<IHeaderTitleControlsProps> = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames(className, 'cssClass[globalHeaderTitleControls]')} {...htmlDivAttributes}>
        {children}
    </div>
);

HeaderTitleControls.displayName = 'HeaderTitleControls';
