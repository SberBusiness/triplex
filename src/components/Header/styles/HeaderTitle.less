@import 'src/styles/style';

.globalHeaderTitle {
    padding: @ufs-theme-page-padding-y @ufs-theme-page-padding-x;
    display: flex;
    justify-content: space-between;

    .headerTitleContent {
        .globalHeaderTitleContentText {
            .font-medium();
            .font-size(@font-size-21);

            text-transform: none;
            color: var(--triplex-Header-Title_Color);
            margin: 0;
            padding: 0;

            outline: none;
            box-sizing: border-box;
            text-transform: initial;
        }

        .headerTitleContentSubhead {
            .font-size(@font-size-14);

            color: var(--triplex-Header-TitleSubhead_Color);
        }
    }

    .globalHeaderTitleControls {
        white-space: nowrap;
        padding-left: @ufs-theme-padding-before-controls-column;
    }
}

.headerLink + .globalHeaderTitle {
    padding-top: 0;
}

@media (max-width: @screen-sm-max) {
    .globalHeaderTitle {
        flex-wrap: wrap;
        padding: 20px 16px;

        .headerTitleContent {
            width: 100%;

            .globalHeaderTitleContentText {
                .font-size(@font-size-16);
            }

            .headerTitleContentSubhead {
                margin-top: 4px;
            }
        }

        .globalHeaderTitleControls {
            padding-left: 0;
            white-space: normal;
            // Отступы, компенсирующие отступы у Button.
            margin-left: -16px;

            > button,
            > .globalButtonDropdownExtended {
                margin-left: 16px;
                margin-top: 16px;
            }
        }
    }
}
