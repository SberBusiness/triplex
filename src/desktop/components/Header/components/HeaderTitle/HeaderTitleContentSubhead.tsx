import * as React from 'react';
import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';

export interface IHeaderTitleContentSubheadProps extends React.HTMLAttributes<HTMLDivElement> {}

/**
 * Подзаголовок.
 */
export const HeaderTitleContentSubhead: React.FC<IHeaderTitleContentSubheadProps> = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames(className, 'cssClass[headerTitleContentSubhead]')} {...htmlDivAttributes}>
        {children}
    </div>
);

HeaderTitleContentSubhead.displayName = 'HeaderTitleContentSubhead';
