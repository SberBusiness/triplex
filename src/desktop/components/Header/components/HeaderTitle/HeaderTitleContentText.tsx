import * as React from 'react';
import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';

interface IHeaderTitleContentTextProps extends React.HTMLAttributes<HTMLHeadingElement> {}

/**
 * Основной заголовок.
 */
export const HeaderTitleContentText: React.FC<IHeaderTitleContentTextProps> = ({children, className, ...HTMLHeadingAttributes}) => (
    <h1
        className={classnames(className, 'cssClass[globalHeaderTitleContentText]')}
        //* При открытии LightBox/ModalWindow - заголовок будет зачитываться VoiceOverом.
        data-first-interaction-element={true}
        tabIndex={-1}
        {...HTMLHeadingAttributes}
    >
        {children}
    </h1>
);

HeaderTitleContentText.displayName = 'HeaderTitleContentText';
