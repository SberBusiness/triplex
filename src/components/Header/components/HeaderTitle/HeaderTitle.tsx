import {HeaderTitleContent} from '@sberbusiness/triplex/components/Header/components/HeaderTitle/HeaderTitleContent';
import {HeaderTitleControls} from '@sberbusiness/triplex/components/Header/components/HeaderTitle/HeaderTitleControls';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';
import React from 'react';

interface IHeaderTitleProps extends React.HTMLAttributes<HTMLDivElement> {}

interface IHeaderTitleFC extends React.FC<IHeaderTitleProps> {
    Content: typeof HeaderTitleContent;
    Controls: typeof HeaderTitleControls;
}

/**
 * Первый уровень Header. Содержит заголовок, подзаголовок и кнопки действий.
 */
export const HeaderTitle: IHeaderTitleFC = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames(className, 'cssClass[globalHeaderTitle]')} {...htmlDivAttributes}>
        {children}
    </div>
);

HeaderTitle.Content = HeaderTitleContent;
HeaderTitle.Controls = HeaderTitleControls;
HeaderTitle.displayName = 'HeaderTitle';
