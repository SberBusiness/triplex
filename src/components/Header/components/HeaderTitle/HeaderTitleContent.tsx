import React from 'react';
import {HeaderTitleContentText} from '@sberbusiness/triplex/components/Header/components/HeaderTitle/HeaderTitleContentText';
import {HeaderTitleContentSubhead} from '@sberbusiness/triplex/components/Header/components/HeaderTitle/HeaderTitleContentSubhead';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';

/** Свойства компонента HeaderTitleContent. */
export interface IHeaderTitleContentProps extends React.HTMLAttributes<HTMLDivElement> {}

/** Часть HeaderTitle с заголовком и подзаголовком. */
export const HeaderTitleContent = Object.assign(
    React.forwardRef<HTMLDivElement, IHeaderTitleContentProps>(function HeaderTitleContent({children, className, ...rest}, ref) {
        return (
            <div className={classnames(className, 'cssClass[headerTitleContent]')} {...rest} ref={ref}>
                {children}
            </div>
        );
    }),
    {
        Subhead: HeaderTitleContentSubhead,
        Text: HeaderTitleContentText,
    }
);

HeaderTitleContent.displayName = 'HeaderTitleContent';
