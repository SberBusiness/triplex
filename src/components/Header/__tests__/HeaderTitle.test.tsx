import React from 'react';
import renderer from 'react-test-renderer';
import {HeaderTitle} from '../components/HeaderTitle/HeaderTitle';

describe('HeaderTitle', () => {
    beforeEach(() => {
        allure.feature('HeaderTitle');
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <HeaderTitle>
                    <HeaderTitle.Content>
                        <HeaderTitle.Content.Text>Text</HeaderTitle.Content.Text>
                        <HeaderTitle.Content.Subhead>Subhead</HeaderTitle.Content.Subhead>
                    </HeaderTitle.Content>
                    <HeaderTitle.Controls>Controls</HeaderTitle.Controls>
                </HeaderTitle>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
