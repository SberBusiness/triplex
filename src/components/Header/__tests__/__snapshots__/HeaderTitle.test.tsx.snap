// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle renders correctly 1`] = `
<div
  className="cssClass[globalHeaderTitle]"
>
  <div
    className="cssClass[headerTitleContent]"
  >
    <h1
      className="cssClass[globalHeaderTitleContentText]"
      data-first-interaction-element={true}
      tabIndex={-1}
    >
      Text
    </h1>
    <div
      className="cssClass[headerTitleContentSubhead]"
    >
      Subhead
    </div>
  </div>
  <div
    className="cssClass[globalHeaderTitleControls]"
  >
    Controls
  </div>
</div>
`;
